tests/test_tasks.py

```diff
@@ -1,6 +1,91 @@
-from ubi_manifest.worker.tasks.dummy_task import dummy_task
+from unittest import mock
 
+from pubtools.pulplib import RpmUnit
 
-def test_dummy_task():
-    result = dummy_task()
-    assert result == "CELERY_OK"
+from ubi_manifest.worker.tasks import depsolve
+
+from .utils import MockLoader, create_and_insert_repo
+
+
+def test_depsolve_task(pulp):
+    with mock.patch(
+        "pubtools.pulplib.YumRepository.get_debug_repository"
```

Comment:
Not sure if I'm missing something, but to me it seems like get_debug_repository should work just fine on the fake pulp client without any need for mocking.

It might be better to do it that way if you can get it working, because then the return types would be more accurate.  I see you made the mock return a repo object directly, but the real method returns a future (which is a proxy future so you can use it as if it were a real repo in most cases, but still a little inaccurate).

Reply:
Yeah I would like to get rid off mocking this but actually this is a bit problematic because pubtools.pulplib uses 'notes.relative_url' for searching related debug repo (and also other related repos). But 'relative_url' field in Repository model class is not mapped with any pulp field and is populated by field - yum_distributor.config.relative_url if the distributor exists. 

I believe it's correct to take relative_url from distributor's config rather than from repo notes.
So I guess one way of fixing this is to search for distributor in the YumRepository._get_related_repo():
`criteria = Criteria.with_field("relative_url", relative_url)
 distributor = client.search_distributor(criteria)
 if distributor:
     return client.get_repostory(distributor.repo_id)
`
Does that look OK?

Reply:
Ah, right.  Yeah, I do believe that the code in pulplib shouldn't depend on notes.relative_url and changing it as in the above example would be a good idea.